### PR TITLE
use convert_to_b64 for system level encoding

### PR DESCRIPTION
**Pull request by jignesh-crest**
## Closes https://github.com/elastic/connectors-py/issues/275
This PR uses on-disk base64 encoding for fetched content, using the method `convert_to_b64` from utils.

#### Pre-Review Checklist
- [x] Tested the changes locally

### CODE REVIEW

**Comment by jignesh-crest** [comment]
buildkite test this

File: connectors/sources/gcs.py [hunk 1]
```diff
@@ -311,8 +313,12 @@ async def get_content(self, blob, timestamp=None, doit=None):
...
             return
         logger.debug(f"Downloading {blob_name}")
-        content = b""
-        async with TemporaryFile("wb+") as async_buffer:
+        document = {
+            "_id": blob["id"],
+            "_timestamp": blob["_timestamp"],
+        }
+        source_file_name = ""
+        async with NamedTemporaryFile(mode="wb+", delete=False) as async_buffer:
```
Comment (tarekziade) [inline connectors/sources/gcs.py:321]: why the `+` ?

File: connectors/sources/gcs.py [hunk 2]
```diff
@@ -325,13 +331,16 @@ async def get_content(self, blob, timestamp=None, doit=None):
                 )
             )
             await async_buffer.seek(0)
```
Comment (tarekziade) [inline connectors/sources/gcs.py:333]: why do you need to keep that seek ? I think you can remove it

File: connectors/sources/gcs.py [hunk 3]
```diff
@@ -325,13 +331,16 @@ async def get_content(self, blob, timestamp=None, doit=None):
...
-            content = await async_buffer.read()
+            source_file_name = async_buffer.name
+        await asyncio.to_thread(
```
Comment (tarekziade) [inline connectors/sources/gcs.py:335]: can you add a logger.debug to say we are calling convert_to_b64

**Review by tarekziade — COMMENTED** [review]
Thanks! just a few nits

File: connectors/sources/gcs.py (hunk 2 above)
Comment (jignesh-crest) [reply connectors/sources/gcs.py:333]: Sure, thanks.

File: connectors/sources/gcs.py (hunk 1 above)
Comment (jignesh-crest) [reply connectors/sources/gcs.py:321]: I forgot to remove it. Thanks for reminding me.

**Comment by tarekziade** [comment]
buildkite test this

**Review by tarekziade — APPROVED** [review]

**Comment by tarekziade** [comment]
buildkite test this

**Pull request merged**